zomes/hc_zome_trust_graph/src/trust_graph.rs

```diff
@@ -0,0 +1,91 @@
+use hdk::prelude::*;
+use hc_zome_trust_atom::*;
+use hc_zome_trust_atom::TrustAtom;
+use std::collections::hash_map;
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, SerializedBytes)]
+struct TrustGraph {
+  trust_atoms: BTreemap<u64, TrustAtom>, //listed by id
+  rollup: TrustAtom, // algorithm from agents you placed TA for given filter
+}
+
+impl TrustGraph {
+    fn create(agents: Vec<EntryHash>, tag_filter: Option<LinkTag>) -> ExternResult<Self> { // TODO: construct agent list for input. Enum?
+        let base_address: EntryHashB64 = agent_info()?.agent_latest_pubkey.into();
+        let links: Vec<Link> = get_links(base_address.clone(), tag_filter.clone())?;
+        let trust_atoms_vec = convert_links_to_trust_atoms(base_address.clone(), links.clone())?;
+        let trust_atoms = BTreeMap::new();
+        for ta in trust_atoms_vec.clone() {
+          trust_atoms.insert(ta.id, ta);
+        }
+        let trust_atom = trust_atoms_vec.pop(); //any TA should give target info.. TODO: verify
+        let algo;
+        let trust_atoms_collection = BTreeMap::new();
+        for agent in agents {
+          let agent_links = get_links(agent.clone(), tag_filter)?;
+          let agent_trust_atoms = convert_links_to_trust_atoms(agent.clone() agent_links)?;
+          trust_atoms_collection.insert(agent.clone(), agent_trust_atoms.clone());
+          let sum;
+          for ta in agent_trust_atoms { // why would there ever be more than one? is there a risk of other agents mislabeling?
+            let float_val = ta.value.parse()?;
+            if float_val > 0 {
+            sum += float_val^1.618; // raised to PHI smooths out weights curve
+            }
+          }
+            }
+          algo += sum;
+        };
+        let rollup = trust_atom::create(
+          trust_atom.target_entry_hash, trust_atom.label, None, algo, trust_atoms_collection
```

Comment:
In my reading of this we only create one TrustAtom, regardless of how many targets there are, of my atoms, and my (agent)atoms' atoms.

What we want is to end up with one rollup TrustAtom for every one of the targets of my atoms, and the targets of my (agent)atoms' atoms.

I added to the example in #25 to try to clarify:

```
eg : given my TAs:

me -[holochain, 0.99]-> zippy
me -[holochain, 0.80]-> robert
me -[holochain, 0.99]-> HIA
me -[engineering, 0.88]-> Telos

zippy -[holochain, 0.99]-> HIA
robert -[holochain, -0.99]-> HIA

robert -[engineering, 0.99]-> Ethereum

then rollup atoms will be:
me -[rollup, holochain, 0.98]-> HIA                  // actual value is TBD
me -[rollup, engineering, 0.99]-> Ethereum     // actual value is TBD
me -[rollup, engineering, 0.88]-> Telos            // actual value is TBD

```